donut/modules/newsgroups/helpers.py

```diff
@@ -203,7 +204,14 @@ def get_owners(group_id):
     """
     with flask.g.pymysql_db.cursor() as cursor:
         cursor.execute(query, group_id)
-        return cursor.fetchall()
+        res = cursor.fetchall()
+        owners = defaultdict(list)
```

Comment:
I would try to avoid unusual Python classes like this. Could just use [`dict.setdefault(row['pos_name'], []).append(...)`](https://docs.python.org/3/library/stdtypes.html#dict.setdefault).